### PR TITLE
Update account artifacts and signer, add signer test

### LINKED ISSUE #91: Replace signer's `hash_multicall` with `tx_info.transaction_hash`
See [OpenZeppelin/cairo-contracts/pull/233](https://github.com/OpenZeppelin/cairo-contracts/pull/233).

### PR DESCRIPTION
**Pull request by andrew-fleming**
This PR proposes to update the account artifact and ABI with the changes merged from Cairo Contracts [#233](https://github.com/OpenZeppelin/cairo-contracts/pull/233). This PR removes `hash_multicall` from the `signer.py` module and utilizes Starknet's `calculate_transaction_hash_common`.

To ensure the viability of the changes to the Account artifact files and Signer module, this PR also proposes to include a test case on the Signer `sign_transaction` method. This test case:
- creates the Account contract instance by building the contract definition from the `Account.json` artifact file
- introduces two test methods (`send_transaction` and `send_transactions`) which format the calls before passing them to the `sign_transaction` method and `__execute__` thereafter
- uses the native `contract.cairo` to test transactions

This resolves #91.

### CODE REVIEW

File: tests/test_signer.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+@pytest.fixture(scope="module")
+def event_loop():
+    return asyncio.new_event_loop()
+
+
+@pytest.mark.xfail(
+    sys.version_info >= (3, 9),
+    reason="Issue in cairo-lang. "
+    "See https://github.com/starkware-libs/cairo-lang/issues/27",
+)
```
Comment (martriay) [inline tests/test_signer.py:60]: should we add a TODO or notice here to know when to remove it?

**Review by martriay — APPROVED** [review]
Very good work! Just to be sure, is there any documentation update missing?

Left a comment otherwise it looks good.

File: tests/test_signer.py (hunk 1 above)
Comment (andrew-fleming) [reply tests/test_signer.py:60]: That's a good idea. I thought the bug StarkWare said they fixed was this frozendict issue, but I'll update the reason^.

Comment (andrew-fleming) [reply tests/test_signer.py:60]: Actually, maybe they did fix the issue...the test dependency uses starknet-devnet which uses Cairo-v0.8.0. Nile's tests won't reflect the change until the devnet is updated

**Comment by andrew-fleming** [comment]
> Just to be sure, is there any documentation update missing?

No, everything should run exactly as it did.

---

There's a conflict with dependencies regarding starknet-devnet. The devnet runs cairo-v0.8.0 and won't recognize the updated import (the same update fix in [Cairo Contracts #260](https://github.com/OpenZeppelin/cairo-contracts/pull/260)).
Setting Cairo to v0.8.1 in tox for tests conflicts with devnet; therefore, this PR proposes to add a try/except on the import in question. With these proposed changes, all tests pass and `nile send...` executes correctly on testnet.

**Review by martriay — APPROVED** [review]

File: tests/test_signer.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        hex(account.contract_address), build_calls, nonce, max_fee
+    )
+    return await account.__execute__(call_array, calldata, nonce).invoke(
+        signature=[sig_r, sig_s]
+    )
+
+
+@pytest.fixture(scope="module")
+def event_loop():
+    return asyncio.new_event_loop()
```
[the remark this answers is not in the document]
Comment (martriay) [reply tests/test_signer.py:52]: lol right

**Pull request merged**